mkosi/qemu.py

```diff
@@ -30,17 +30,38 @@
 from mkosi.run import MkosiAsyncioThread, run, spawn
 from mkosi.tree import copy_tree, rmtree
 from mkosi.types import PathString
-from mkosi.util import (
-    INVOKING_USER,
-    StrEnum,
-    qemu_check_kvm_support,
-    qemu_check_vsock_support,
-)
+from mkosi.util import INVOKING_USER, StrEnum
 
 
 class QemuDeviceNode(StrEnum):
+    kvm = enum.auto()
     vhost_vsock = enum.auto()
 
+    def device(self) -> Path:
+        return Path("/dev") / str(self)
+
+    def description(self) -> str:
+        return {
+            QemuDeviceNode.kvm: "KVM acceleration",
+            QemuDeviceNode.vhost_vsock: "a VSock device",
+        }[self]
```

Comment:
If there's no other users for the value, then what `description` does here is in theory the job of `__str__`, to make a nice user-visible string.

I don't think there's any users of the automatic value of `vhost-vsock` for the `vhost_vsock` member, so if you changed the values to `KVM acceleration` and `a VSock device` all the calls to `description` in the f-strings would be obselete.

Reply:
Hmm, this really is a description and not really a stringification to me, so reusing __str__() for that doesn't feel right. Also we do use the stringification already in the available() method.